tgui/packages/tgui/interfaces/Apc/index.js

```diff
@@ -0,0 +1,260 @@
+    }
+  };
+
+  const hasPermission = () => {
+    if (is_ai || is_silicon || can_access_remotely) {
+      return aidisabled ? false : true;
+    }
+    return locked ? false : true;
+  };
+
+  const isLocalAccess = () => {
+    return setup_networkapc < 2 && !can_access_remotely;
+  };
+
+  const hostConnectionDisplay = () => {
+    if (isLocalAccess()) {
+      return null;
+    } else {
+      return (
+        <LabeledList.Item label="Host Connection:">
+          <font color={host_connected ? "green" : "red"}>{host_connected ? "OK" : "NONE"}</font>
+        </LabeledList.Item>
+      );
+    }
+  };
+
+  const renderPoweredAreaApc = () => {
+    return (
+      <Window title="Area Power Controller" width={400} height={data["wiresexposed"] ? 680 : 420}>
+        <Window.Content>
+          <Section title={"Area Power Controller (" + area_name + ")"}>
+            {isLocalAccess() ? <Box align="center" bold fill>Swipe ID card to {locked ? "unlock" : "lock"} interface</Box> : null}
+            {isLocalAccess() ? <Divider /> : null}
+            <LabeledList>
+              <LabeledList.Item label="Main Breaker">
+                <Button content="Off" disabled={!hasPermission() && operating} onClick={() => { onOperatingChange(OFF); }} selected={!operating} />
+                <Button content="On" disabled={!hasPermission() && !operating} onClick={() => { onOperatingChange(ON); }} selected={operating} />
+              </LabeledList.Item>
+              {cellDisplay()}
+              <LabeledList.Item label="External Power">
+                <Box>{mainStatusToText()}</Box>
+              </LabeledList.Item>
+            </LabeledList>
+          </Section>
+          <Section title="PowerChannel">
+            <LabeledList>
+              <PowerChannelSection powerChannel={POWER_CHANNEL_EQUIPMENT} />
+              <PowerChannelSection powerChannel={POWER_CHANNEL_LIGHTING} />
+              <PowerChannelSection powerChannel={POWER_CHANNEL_ENVIRONMENTAL} />
+              <LabeledList.Item label="Total Load">
+                <Box>{lastused_total} W</Box>
+              </LabeledList.Item>
+            </LabeledList>
+          </Section>
+          <Section>
+            <Stack>
+              <Stack.Item align="center">
+                <Box>Cover lock:</Box>
+              </Stack.Item>
+              <Stack.Item align="center">
+                {coverLockDisplay()}
+              </Stack.Item>
+            </Stack>
+            {can_access_remotely ? <Button content="Overload lighting circuit" onClick={() => { onOverload(); }} /> : null}
+          </Section>
+          {wiresexposed && !is_ai ? <AccessPanel /> : null}
+        </Window.Content>
+      </Window>
+    );
+  };
+
+  const renderUnPoweredAreaApc = () => {
+    return (
+      <Window title="Area Power Controller" width={400} height={wiresexposed ? 500 : 350}>
+        <Window.Content>
+          <Section title={"Area Power Controller (" + area_name + ")"}>
```

Comment:
x

Reply:
assuming this is just referencing location of second occurrance?